src/main/kotlin/com/jayrave/moshi/pristineModels/FieldMappingImpl.kt

```diff
@@ -24,29 +22,7 @@ internal class FieldMappingImpl<in T : Any, F>(
         }
     }
 
-
-    @Throws(ValueCanNotBeNullException::class)
-    fun lastReadValueInCurrentThread(): F {
-        // It seems that even if [F] is a non-null type, this method will happily return a
-        // `null` value (they are no nullability checks I could see in the decompiled code.
-        // I guess because of erasure there isn't enough information). Therefore do a manual
-        // check & throw if required
-
-        val value = readValues.get()
-
-        @Suppress("UNCHECKED_CAST")
-        return when (valueCanBeNull) {
```

Comment:
Is there a use case for exposing the property `valueCanBeNull`?  Can we not assume that properties declared as `non-null` should have value and through an _Exception_ ALWAYS?

Reply:
> properties declared as `non-null`

`valueCanBeNull` is that flag that talks about the intention of whether something can be null or not which we automatically compute when `field` is used to create a `FieldMapping`. I might be misunderstanding your question @mbalajee - please let me know if that's the case

Reply:
I guess this was relevant to have when `lastReadValueInCurrentThread` was around. Since that method isn't present anymore we could potentially let this go but now I wonder if we should instead check this in `read` instead. Thoughts @mbalajee ?

Reply:
> Ah! I get it now. Yeah, the extn function that's present to easily create a [field already does this](https://github.com/jayrave/moshi-pristine-models/blob/aa60b0e4ff55518a6d5022afd9ec9aa404833783/src/main/kotlin/com/jayrave/moshi/pristineModels/MapperExtn.kt#L25) (automatically figuring out if a field can carry null value or not). `valueCanBeNull` is still exposed since I wanted this library to also be usable Java too where this value will have to passed in as we can't automatically figure it out:
> 
> ```java
> class UserMapper : Mapper<User>() {
>     val name = field("user_field", false, object : PropertyExtractor<User, String> {
>         override val type: Type = String::class.javaObjectType
>         override fun extractFrom(t: User): String = t.getName()
>     })
> 
>     val age = field("user_age", false, object : PropertyExtractor<User, Int> {
>         override val type: Type = Int::class.javaPrimitiveType!!
>         override fun extractFrom(t: User): Int = t.getAge()
>     })
> 
>     override fun create(value: Value<User>): User {
>         return User(value of name, value of age)
>     }
> }
> ```

Got it.  `@Nullable` annotation (`valueCanBeNull = property.annotations.any { it is Nullable }`) is an option in Java.  But i guess you wanted this to be annotation free.